src/SharedUtilities/implicit_functions.jl

```diff
@@ -0,0 +1,125 @@
+### These functions are here for now, but linsolve!, ldiv!, and
+### thomas_algorithm! will soon be moved to ClimaCore.
+
+import LinearAlgebra
+import ClimaCore: Spaces
+
+export linsolve!, ldiv!, thomas_algorithm!
+
+# Function required by OrdinaryDiffEq.jl
+linsolve!(::Type{Val{:init}}, f, u0; kwargs...) = _linsolve!
+_linsolve!(x, A, b, update_matrix = false; kwargs...) =
+    LinearAlgebra.ldiv!(x, A, b)
+
+# Function required by Krylov.jl (x and b can be AbstractVectors)
+# See https://github.com/JuliaSmoothOptimizers/Krylov.jl/issues/605 for a
+# related issue that requires the same workaround.
+function LinearAlgebra.ldiv!(x, A::AbstractTridiagonalW, b)
+    A.temp1 .= b
+    LinearAlgebra.ldiv!(A.temp2, A, A.temp1)
+    x .= A.temp2
+end
+
+function LinearAlgebra.ldiv!(
+    x::Fields.FieldVector,
+    A::AbstractTridiagonalW,
+    b::Fields.FieldVector,
+)
+    _ldiv!(x, A, b, axes(x.soil.ϑ_l))
```

Comment:
Perhaps better to move into Soil since this has hardcoded theta variables?

Im not totally sure how this will generalize to a case with more prognostic variables. I guess it will be handled by ClimaTimesteppers, but as we continue to develop here, prior to this being moved to CTS, Im not sure what is best. For example, if we start testing the full soil implicit problem, or even the richards+canopy problem (canopy prognostic variables present), Im not sure if this will run.

Reply:
Hmm this is a good point. Maybe we could recursively unpack the properties of `x` like we do in the new `dss!` implementation. Then we could choose one of the properties and dispatch off of its space for the call to `_ldiv!`.

Reply:
sure! does it make sense to do this in a different PR? maybe it will be clearer what to do are running the implicit solver?

Reply:
Yeah, I think doing this in a different PR is a good idea. I'll open an issue and we can address it soon, maybe once we start using the implicit solver for RRE.